File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubeTrackDetailsLoader.java
package com.sedmelluq.discord.lavaplayer.source.youtube;

import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface;

public interface YoutubeTrackDetailsLoader {
  YoutubeTrackDetails loadDetails(HttpInterface httpInterface, String videoId, boolean requireFormats);
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubeTrackDetailsLoader.kt
package com.sedmelluq.discord.lavaplayer.source.youtube

import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface

interface YoutubeTrackDetailsLoader {
    fun loadDetails(
        httpInterface: HttpInterface,
        videoId: String,
        requireFormats: Boolean,
        sourceManager: YoutubeItemSourceManager
    ): YoutubeTrackDetails?
}
